src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsWindowMapping.java

```diff
@@ -69,6 +86,58 @@ public static LocalDateTime getFirstDateTimeMustBe() {
         return null;
     }
 
+    @NotNull
+    private static LocalDate handleAdjournedHearingType(SscsCaseData caseData) {
+        String dateType = caseData.getAdjournCaseNextHearingDateType();
+        if (FIRST_AVAILABLE_DATE.equalsIgnoreCase(dateType)) {
+            return LocalDate.now().plusDays(DAYS_TO_ADD_HEARING_WINDOW_TODAY_ADJOURNMENT);
+        } else if (FIRST_AVAILABLE_DATE_AFTER.equalsIgnoreCase(dateType)) {
+            return calculateDateOfAdjournedHearing(caseData);
+        } else {
+            throw new IllegalArgumentException("Unexpected adjournCaseNextHearingDateType: "
```

Comment:
You should include your case id in errors and log messages.

Reply:
done